data_model/src/query.rs

```diff
@@ -732,6 +734,21 @@ pub mod asset {
         type Output = bool;
     }
 
+    query! {
+        /// [`FindAssetRegisteringTransaction`] Iroha Query gets [`AssetDefinitionId`] as input and finds
+        /// [`Hash`] of the transaction that registered this asset.
+        #[derive(Display)]
+        #[display(fmt = "Find asset registering transaction of `{id}` asset")]
+        pub struct FindAssetRegisteringTransaction {
+            /// `Id` of an [`AssetDefinition`] .
+            pub id: EvaluatesTo<AssetDefinitionId>,
+        }
+    }
+
+    impl Query for FindAssetRegisteringTransaction {
+        type Output = iroha_crypto::Hash;
```

Comment:
```suggestion
        type Output = HashOf<VersionedCommittedBlock>;
```

Why not like so?

Reply:
I think you mean:
```suggestion
        type Output = HashOf<VersionedSignedTransaction>;
```
 Are we finding transaction or block?

Reply:
It is indeed a transaction, yes.